python.d/python_modules/bases/FrameworkServices/SocketService.py

```diff
@@ -249,6 +273,21 @@ def _parse_config(self):
             except (KeyError, TypeError):
                 self.debug('No port specified. Using: "{0}"'.format(self.port))
 
+            if _SSL_SUPPORT:
+                try:
+                    self.ssl = bool(self.configuration['ssl'])
+                except (KeyError, TypeError):
+                    self.debug('No SSL preference specified, not using SSL.')
+                    self.ssl = False
+                else:
+                    try:
+                        self.key = str(self.configuration['ssl_key'])
+                        self.cert = str(self.configuration['ssl_cert'])
+                    except (KeyError, TypeError):
+                        self.debug('No SSL client certificate configuration found.')
+                        self.key = None
+                        self.cert = None
+
```

Comment:
I suggest to avoid stacking `try/except` blocks. 

I think this is enough: (yeah, `SocketService` should be rewritten)
```python
 if _SSL_SUPPORT:
   self.key = self.configuration['ssl_key']
   self.cert = self.configuration['ssl_cert']
```

Reply:
The intent here was to avoid wasting time with the client certificate stuff if we're not using SSL, but still provide slightly more specific debug info (there's a pretty big difference between not using SSL at all and not having a client certificate), though looking back at it it's kind of non-obvious what it's doing (`else` clauses on `try`/`except` blocks are kind of rarely used).

I think I'm going to just rework this whole section anyway so that it doesn't silently ignore a request to use SSL if there is no `ssl` module.

Reply:
i mean simple `self.key = self.configuration.get('ssl_key')` does same work as `try/except`
you can use `self.debug` ofc if you need
```python
 if _SSL_SUPPORT:
   self.key = self.configuration.get('ssl_key')
   if not self.key:
       self.debug("......")
   self.cert = self.configuration.get('ssl_cert')
```

Reply:
Ah, OK, that makes much more sense.  Kind of surprised I didn't realize that.